Docker/test/selenium/specs/quickstatements/quickstatements.js

```diff
@@ -30,28 +41,145 @@ describe( 'QuickStatements Service', function () {
+		browser.executeQuickStatement( 'Q1|DSv|"Kommer det funka?"' );
 
-		// click run
-		$( "button[tt='run']" ).waitForDisplayed();
-		$( "button[tt='run']" ).click();
+		// go look at wikibase
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+
+		assert( _.isEmpty( responseQ1.data.entities.Q1.descriptions ) !== true );
+	} );
 
-		// wait for the creating
-		browser.pause( 5000 );
+	it.skip( 'Should be able to add a sitelink to an item', function () {
+
+		browser.executeQuickStatement( 'Q1|Sclient_wiki|"Main_Page"' );
 
 		// go look at wikibase
-		browser.url( process.env.MW_SERVER + '/wiki/Item:Q1' );
-		$( '.wikibase-toolbarbutton.wikibase-toolbar-item.wikibase-toolbar-button.wikibase-toolbar-button-add' ).waitForDisplayed();
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+
+		assert( _.isEmpty( responseQ1.data.entities.Q1.sitelinks ) !== true );
+	} );
+
+	it( 'Should be able to add a property to an item', function () {
+
+		propertyId = browser.call( () => WikibaseApi.getProperty( 'string' ) );
+
+		browser.executeQuickStatement( 'Q1|' + propertyId + '|"Will it blend?"' );
+
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+		assert( responseQ1.data.entities.Q1.claims[ propertyId ][ 0 ].type === 'statement' );
+
+	} );
+
+	it( 'Should be able to add statement with qualifiers', function () {
+
+		propertyIdItem = browser.call( () => WikibaseApi.getProperty( 'wikibase-item' ) );
+
+		browser.executeQuickStatement( 'Q1|' + propertyIdItem + '|Q1|' + propertyIdItem + '|Q1' );
+
+		const responseQ1 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q1.json' );
+		assert( responseQ1.data.entities.Q1.claims[ propertyId ][ 0 ].type === 'statement' );
+
+	} );
+
+	it( 'Should be able to add a property with "wikibase-item" reference', function () {
+
+		propertyIdItem = browser.call( () => WikibaseApi.getProperty( 'wikibase-item' ) );
+		const propertyNumber = propertyIdItem.replace( 'P', '' );
+		browser.executeQuickStatement( 'Q2|' + propertyIdItem + '|Q2|S' + propertyNumber + '|Q2|S' + propertyNumber + '|Q2' );
+
+		const responseQ2 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q2.json' );
+		assert( ( propertyIdItem in responseQ2.data.entities.Q2.claims ) === true );
+	} );
+
+	it( 'Should be able to add a property with "url" reference', function () {
+
+		propertyURL = browser.call( () => WikibaseApi.getProperty( 'url' ) );
+		const url = '"https://www.wikidata.org"';
+		const propertyNumber = propertyURL.replace( 'P', '' );
+		browser.executeQuickStatement( 'Q2|' + propertyIdItem + '|Q3|S' + propertyNumber + '|' + url );
+
+		const responseQ2 = browser.makeRequest( process.env.MW_SERVER + '/wiki/Special:EntityData/Q2.json' );
+		assert( ( propertyURL in responseQ2.data.entities.Q2.claims ) === true );
```

Comment:
This fails, as propertyURL is only used in a reference.

Reply:
Why shouldn't I be able to use it only as a reference?

This query( Q2|P2|Q3|S3|"https://www.wikidata.org" ) works but for some reason not in this test. If i login after the test suite is run (also if i skip the last merge test) and run this query it succeeds.

Reply:
Ok ignore this the item it's referencing is created in the below test.

Reply:
You meant the assertion. jesus f-ing christ im stupid.